palantir-java-format-native/build.gradle

```diff
@@ -0,0 +1,111 @@
+        classpath 'com.palantir.gradle.utils:platform:0.10.0'
+    }
+}
+
+
+apply plugin: 'java-library'
+apply plugin: 'org.graalvm.buildtools.native'
+apply plugin: 'com.palantir.external-publish-custom'
+
+dependencies {
+    implementation project(':palantir-java-format')
+}
+
+import com.palantir.platform.OperatingSystem
+import com.palantir.platform.Architecture
+
+def operatingSystem = OperatingSystem.get()
+def architecture = Architecture.get()
+
+def exports = [
+        'jdk.compiler/com.sun.tools.javac.file',
+        'jdk.compiler/com.sun.tools.javac.main',
+        'jdk.compiler/com.sun.tools.javac.parser',
+        'jdk.compiler/com.sun.tools.javac.tree',
+        'jdk.compiler/com.sun.tools.javac.util',
+        'jdk.compiler/com.sun.tools.javac.code',
+        'jdk.compiler/com.sun.tools.javac.api'
+]
+
+def jvmArgList = exports.collect { value -> "--add-exports=${value}=ALL-UNNAMED".toString() }
+
+
+graalvmNative {
+    binaries {
+        main {
+            javaLauncher = javaToolchains.launcherFor {
+                languageVersion = JavaLanguageVersion.of(23)
+                vendor = JvmVendorSpec.GRAAL_VM
+            }
+            imageName = String.format('palantir-java-format-%s-%s', operatingSystem.uiName(), architecture.uiName())
+            mainClass = 'com.palantir.javaformat.java.Main'
+            fallback = false
+            sharedLibrary = false
+            classpath.setFrom(configurations.named('runtimeClasspath'))
+            configurationFileDirectories.from(file('src/main/resources/META-INF/native-image'))
+            buildArgs.addAll("-H:ConfigurationFileDirectories=${projectDir}/src/main/resources/META-INF/native-image",
+                    '-H:+UnlockExperimentalVMOptions',
+                    '-H:IncludeResourceBundles=com.sun.tools.javac.resources.compiler',
+                    '-H:IncludeResourceBundles=com.sun.tools.javac.resources.javac',
+                    '--no-fallback',
+                    '--initialize-at-build-time=com.sun.tools.javac.file.Locations',
+                    '-H:+ReportExceptionStackTraces',
+                    '-H:-UseContainerSupport',
+                    "-march=compatibility")
+            jvmArgs.addAll(jvmArgList)
+        }
+    }
+}
+
+
+// The image requires parameters to be passed to; eg. gw nativeRun -Pargs=--palantir,-i,<file>
+tasks.named("nativeRun", org.graalvm.buildtools.gradle.tasks.NativeRunTask) {
+    runtimeArgs.addAll(project.hasProperty('appArgs') ? project.appArgs.split(',') : [])
+}
+
+def osExtension = operatingSystem == OperatingSystem.MACOS ? 'bin' : 'exe'
+
+externalPublishing {
+    publishing {
+        publications {
+            nativeImage(MavenPublication) {
+                fileTree("build/native/nativeCompile", { it.include("**/*") }).forEach { file ->
+                    artifact(file) {
+                        classifier = String.format("nativeImage-%s", getOsArchName(file))
+                        extension = osExtension
```

Comment:
Shouldn't this vary per artifact? Currently it seems to be fixed based on the OS running the publish job?